src/movies.js

```diff
@@ -1,28 +1,138 @@
+  const directors = movies.map((movie) => movie.director);
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(bestDirector) {
+  return bestDirector.filter(
+    (stevenSpielberg) => 
+      stevenSpielberg.director === "Steven Spielberg" && stevenSpielberg.genre.includes("Drama")).length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(averageMovies) {
+  if (!averageMovies.length){
+    return 0;
+  }
+
+  let total = averageMovies.reduce((a, b) => {
+    if (b.rate) {
+      return a + b.rate;
+    } else {
+      return a;
+    }
+  }, 0);
+  return Number((total / averageMovies.length).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(dramaMovies) {
+  let moviesInspector = dramaMovies.filter((theMovie) =>
+    theMovie.genre.includes("Drama")
+  );
+  return scoresAverage(moviesInspector);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(scoresAverage) {
+  let arrayOrdered = [];
+  arrayOrdered.sort((a,b) =>{
+    if (a.year > b.year){
+      return 1;
+    } else if (b.year > a.year){
+      return -1;
+    } else {
+      if (a.title > b.title){
+        return 1;
+      } else if (b.title> a.title){
+        return -1;
+      }
+      return 0;
+    }
+  });
+  return arrayOrdered;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(scoresAverage) {
+  return scoresAverage.sort((a, b) => {
+    if (a.title >b.title) {
+      return 1;
+    } else if (a.title < b.title){
+      return -1;
+    } else {
+      return 0;
+    }
+  })
+  .map((stevenSpielberg) => stevenSpielberg.title)
+  .slice(0, 20);
+}
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes() {}
+function convertionHours(hString) {
+  let calcHour = hString.split("h");
+  return calcHour[0] * 60;
+}
+
+function convertionMinutes(minString){
+  let calcMin = minString.split("min");
+  return Number(calcMin[0]);
+}
+
+function convertionDuration(duration) {
+  let timeSpaces = duration.split(" ");
+  let minutes = timeSpaces.reduce((sum, oneSpace) => {
+    if (oneSpace.includes("h")){
+      return sum + convertionHours(oneSpace);
+    }
+    return sum + convertionMinutes(oneSpace);
+  }, 0);
+  return minutes;
+}
+
+function turnHoursToMinutes(movies) {
+  let centArray = movies.map((oneMovie) => {
+    let newMovie = {};
+    newMovie.title = oneMovie.title;
+    newMovie.year = oneMovie.year;
+    newMovie.director = oneMovie.director;
+    newMovie.duration = converDuration(oneMovie.duration);
```

Comment:
en esta línea estáis intentando usar un método `converDuration ` que no existe, se llama `convertionDuration`. Con este cambio ya pasan.